File: app/src/main/java/com/github/qezt/phonenumberidentifier/OnDebouncedClickListener.java
package com.github.qezt.phonenumberidentifier;


import android.os.SystemClock;
import android.view.View;

public abstract class OnDebouncedClickListener implements View.OnClickListener {
    private long coolDown;
    private long lastClickTime;

    public OnDebouncedClickListener() {
        this(300);
    }
    public OnDebouncedClickListener(int coolDown) {
        this.coolDown = coolDown;
    }

    @Override
    public void onClick(View v) {
        long now = SystemClock.elapsedRealtime();
        if (now - lastClickTime < coolDown) return;
        lastClickTime = now;
        onDebouncedClickClick();
    }

    public abstract void onDebouncedClickClick();
}

File: app/src/main/java/com/github/qezt/phonenumberidentifier/OnDebouncedClickListener.kt
package com.github.qezt.phonenumberidentifier


import android.os.SystemClock
import android.view.View

abstract class OnDebouncedClickListener(private val coolDown: Long = 300) : View.OnClickListener {
    private var lastClickTime: Long = 0

    override fun onClick(v: View) {
        val now = SystemClock.elapsedRealtime()
        if (now - lastClickTime < coolDown) return
        lastClickTime = now
        onDebouncedClickClick()
    }

    abstract fun onDebouncedClickClick()
}
